dataland-frontend/src/components/forms/CreateEUTaxonomyForNonFinancials.vue

```diff
@@ -1,139 +1,452 @@
-import DataPointFormElement from "@/components/forms/DataPointFormElement.vue";
 import { defineComponent, inject } from "vue";
+import { useRoute } from "vue-router";
 import Keycloak from "keycloak-js";
 import { assertDefined } from "@/utils/TypeScriptUtils";
+import { getHyphenatedDate } from "@/utils/DataFormatUtils";
+
+import {
+  euTaxonomyPseudoModelAndMappings,
+  euTaxonomyKpiInfoMappings,
+  euTaxonomyKpiNameMappings,
+} from "@/components/forms/parts/kpiSelection/EuTaxonomyPseudoModelAndMappings";
+import { CompanyAssociatedDataEuTaxonomyDataForNonFinancials } from "@clients/backend";
+import { UPLOAD_MAX_FILE_SIZE_IN_BYTES } from "@/utils/Constants";
+import { checkCustomInputs } from "@/utils/ValidationsUtils";
+import { modifyObjectKeys, objectType, updateObject } from "@/utils/UpdateObjectUtils";
+import { formatBytesUserFriendly } from "@/utils/NumberConversionUtils";
+import { ExtendedCompanyReport, ExtendedFile, WhichSetOfFiles } from "@/components/forms/Types";
+import JumpLinksSection from "@/components/forms/parts/JumpLinksSection.vue";
 
 export default defineComponent({
   name: "CreateEUTaxonomyForNonFinancials",
-  components: { DataPointFormElement, FailedUpload, Card, FormKit, SuccessUpload },
+  components: {
+    JumpLinksSection,
+    Calendar,
+    UploadFormHeader,
+    PrimeButton,
+    UploadReports,
+    BasicInformationFields,
+    KPIfieldSet,
+    FailedUpload,
+    Card,
+    FormKit,
+    SuccessUpload,
+  },
   setup() {
     return {
       getKeycloakPromise: inject<() => Promise<Keycloak>>("getKeycloakPromise"),
     };
   },
 
   data: () => ({
-    innerClass: {
-      "formkit-inner": false,
-      "p-inputwrapper": true,
-    },
-    inputClass: {
-      "formkit-input": false,
-      "p-inputtext": true,
-      "w-full": true,
+    formInputsModel: {} as CompanyAssociatedDataEuTaxonomyDataForNonFinancials,
+    fiscalYearEnd: "" as Date | "",
+    convertedFiscalYearEnd: "",
+    reportingPeriod: new Date(),
+    filesToUpload: [] as ExtendedFile[],
+    uploadFiles: [] as ExtendedCompanyReport[],
+    onThisPageLinks: [
+      { label: "Upload company reports", value: "uploadReports" },
+      { label: "Basic information", value: "basicInformation" },
+      { label: "Assurance", value: "assurance" },
+      { label: "CapEx", value: "capex" },
+      { label: "OpEx", value: "opex" },
+      { label: "Revenue", value: "revenue" },
+    ],
+    elementPosition: 0,
+    route: useRoute(),
+    editMode: false,
+    waitingForData: false,
+    checkCustomInputs,
+    formatBytesUserFriendly,
+    maxFileSize: UPLOAD_MAX_FILE_SIZE_IN_BYTES,
+    euTaxonomyKPIsModel: euTaxonomyPseudoModelAndMappings,
+    euTaxonomyKpiNameMappings,
+    euTaxonomyKpiInfoMappings,
+    reportingPeriodYear: new Date().getFullYear(),
+    assuranceData: {
+      None: humanizeString("None"),
+      LimitedAssurance: humanizeString("LimitedAssurance"),
+      ReasonableAssurance: humanizeString("ReasonableAssurance"),
```

Comment:
This should be consistent with "assuranceData" in `CreateEuTaxonomyFOrFinancials.vue`